### PR TITLE
chore: CRW-3196 delete yarn.lock and regen...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore: [CRW-3196](https://issues.redhat.com//browse/CRW-3196) delete yarn.lock and regen with:
```
rm -f yarn.lock tools/build/yarn.lock; yarn install --mode update-lockfile; cp yarn.lock tools/build/yarn.lock
```
Need duplicate yarn.lock so cachito can find and use it; also need to include yarn v1 binary. 

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-3196

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .linelint.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ignore:
+  - .yarn/
```
Comment (svor) [inline .linelint.yml:2]:
```suggestion
  - .yarn/

rules:
  # checks if file ends in a newline character
  end-of-file:
    # set to true to enable this rule
    enable: true
```

Comment (svor) [reply .linelint.yml:2]: if you want to use .linelint.yaml file, you have to specify rules in it

**Review by svor — APPROVED** [review]
LGTM

**Comment by svor** [comment]
Can we copy yarn.lock into `tools/build/` during the cachito build?

**Comment by nickboldt** [comment]
> Can we copy yarn.lock into `tools/build/` during the cachito build?

 flow needs to be one of two things:
* put everything upsteam we need in downstream for the npmjs builds, and use cachito to pull sources and yarn deps
* DON'T put everything in upstream... patch in midsteam via sync.sh script... do the yarn install in midstream and commit all the node_modules/ to downstream (no cachito for building npmjs generators)

the latter approach is bloatier and requires more custom sync scripting
the former is cleaner as long as we:
* ensure that updates are done for every dependabot update to the root .lock file... perhaps have a GH action that fires every time .lock file to root is updated and copied to the subfolder? Seems a trivial thing to write, no?

**Comment by svor** [comment]
ok, let's create a GH action that will be executed when root yarn.lock is changed and copy/replace it in tools/build folder

**Comment by nickboldt** [comment]
replaced by https://github.com/eclipse-che/che-plugin-registry/pull/1634

**Pull request closed without merging**